python/kwiver/sprokit/adapters/adapter_data_set.cxx

```diff
@@ -0,0 +1,147 @@
+
+    .def("empty", &adapter_data_set::empty)
+
+    // get_port_data is also templated
+    .def("get_port_data", [] (adapter_data_set &self, ::sprokit::process::port_t const& port)
+    {
+      object dat = none();
+      kwiver::vital::any const any = self.get_port_data<kwiver::vital::any>(port);
+      dat = kwiver::vital::any_cast<object>(any);
+      return dat;
+    })
+    .def("get_port_data_int", &adapter_data_set::get_port_data<int>)
+    .def("get_port_data_float", &adapter_data_set::get_port_data<float>)
+    .def("get_port_data_string", &adapter_data_set::get_port_data<std::string>)
+    // Next shared ptrs to kwiver vital types
+    .def("get_port_data_image_container", &adapter_data_set::get_port_data<std::shared_ptr<kwiver::vital::image_container > >)
+    .def("get_port_data_descriptor_set", &adapter_data_set::get_port_data<std::shared_ptr<kwiver::vital::descriptor_set > >)
+    .def("get_port_data_detected_object_set", &adapter_data_set::get_port_data<std::shared_ptr<kwiver::vital::detected_object_set > >)
+    .def("get_port_data_track_set", &adapter_data_set::get_port_data<std::shared_ptr<kwiver::vital::track_set > >)
+    .def("get_port_data_feature_track_set", &adapter_data_set::get_port_data<std::shared_ptr<kwiver::vital::feature_track_set > >)
+    .def("get_port_data_object_track_set", &adapter_data_set::get_port_data<std::shared_ptr<kwiver::vital::object_track_set > >)
+    //Next shared ptrs to native C++ types
+    .def("get_port_data_double_vector", &adapter_data_set::get_port_data<std::shared_ptr<std::vector<double > > >)
+    .def("get_port_data_string_vector", &adapter_data_set::get_port_data<std::shared_ptr<std::vector<std::string > > >)
+    .def("get_port_data_uchar_vector", &adapter_data_set::get_port_data<std::shared_ptr<std::vector<unsigned char > > >)
+    //Next kwiver vital types
+    .def("get_port_data_bounding_box", &adapter_data_set::get_port_data<kwiver::vital::bounding_box_d>)
+    .def("get_port_data_timestamp", &adapter_data_set::get_port_data<kwiver::vital::timestamp>)
+    .def("get_port_data_corner_points", &adapter_data_set::get_port_data<kwiver::vital::geo_polygon>)
+    .def("get_port_data_f2f_homography", &adapter_data_set::get_port_data<kwiver::vital::f2f_homography>)
+    ;
+    ads.doc() = R"(
+        Python bindings for kwiver::adapter::adapter_data_set
+
+        Example:
+            >>> from kwiver.sprokit.adapters import adapter_data_set
+            >>> # Following ads has type "data". We can add/get data to/from ports
+            >>> ads = adapter_data_set.AdapterDataSet.create()
+            >>> assert ads.type() == adapter_data_set.DataSetType.data
+            >>> # Can add as a general python object
+            >>> ads.add_value("port1", "a_string")
+            >>> # Can also add by specifying type
+            >>> ads.add_int("port2", 5)
+            >>> # Get both values
+            >>> assert ads.get_port_data("port1") == "a_string"
```

Comment:
Yeah, Python `__getitem__` and `__setitem__` would be really useful for this class. Making this as dict-like as possible would make it much more Pythonic.

Reply:
Added in 1c40fdd8f27f207f3620e362328f94bf75ee9e11